### DISCUSSION TITLE
Error : Failed to save ipynb, cannot read properties of undefined

### DISCUSSION THREAD

**Discussion by MacherelR** [microsoft/vscode-jupyter#9666] [Questions and Answers]
Hi, 
I'm currently using vscode with wsl to code in python. this also uses the vscode-jupyter extension. My notebook works fine but sometimes when I do try to save it, I get the following message : 
Failed to save 'doc.ipynb' : Cannot read properties of undefined (reading 'logger').
Does anyone know why this happens ?

Thanks in advance
Rémy

**Comment by IanMatthewHuff** [comment]
@MacherelR Quick question for you. When this happens, could you open Help > Toggle Developer Tools and see if there is a full stack trace for this error? Would be helpful in diagnosing this.

**Reply by rlaker** [reply]
I recently got this error and this is what the console says (I do not know how to get the "full stack trace").

![image](https://user-images.githubusercontent.com/20686171/166688561-bc505735-e162-49f5-8de3-1de25f26f2d6.png)

**Comment by ghost** [comment]
I had the same problem. The following worked for me:

1. I copied the changes I had made to another file as a backup. 
2. I tried to save my original file again. I elected the option to discard when I got the error again. 
3. The original file was intact with my changes, and I was able to save my file now.
